### DISCUSSION TITLE
Regarding the specification of the function to obtain the ID of the component placed in the schematic.

### DISCUSSION THREAD

**Discussion by homma26153190** [ansys/pyaedt#4650] [Q&A] [answered]
Hello, PyAEDT Experts.
I apologize for taking your time, but could you please share any information regarding the following?

Using PyAEDT, I work with TouchStoneFile. To obtain the ID of the Nport component placed in the schematic, 
I extract component_info from .modeler.schematic.components under Circuit, gathering the following details:
・ID
・InstanceName
・Component
I have attached the sample source code and TouchStone. It worked correctly up to version 0.6.86, but as of version 0.9 (dated 2024/05/07), it no longer retrieves the component information, including ID. Could you inform me of any changes in the related API specifications?

[sample_files.zip](https://github.com/ansys/pyaedt/files/15237338/sample_files.zip)

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @homma26153190 ,

These methods has been deprecated months ago, and finally in 0.9.0 release, they are not maintained anymore.
```
    with Desktop(specified_version="2023.1", non_graphical=False, new_desktop_session=True, close_on_exit=True):
        aedtapp = Circuit("AutomatedProject")

        work_dir = os.getcwd()
        tsfile_path = os.path.join(work_dir, tsfile)
        filename = tsfile.split(".")[0]
    
        comp = aedtapp.modeler.components.create_touchstone_component(tsfile_path)
        
        if any(comp.component_info):
            instance_name = comp.component_info['InstanceName']
            component_name = comp.component_info['Component']
            refdes_value = comp.component_info['RefDes']
            id_value = comp.component_info['ID']
            print('-----')
            print('InstanceName: ' + instance_name)
            print('ComponentName: ' + component_name)
            print('refDes: ' + refdes_value)
            print('ID: ' + id_value)
            print('All infomation: ' + str(comp.component_info))
        else:
            print("Fail")

        aedtapp.modeler.components.add_pin_iports(filename, id_value)
        
        # You need to create a Setup
        
        #----- Solve    
        aedtapp.analyze()

        #----- Release AEDT instance
        aedtapp.release_desktop(close_projects=True,close_desktop=True)
```

There is a method, that can do almosth the same in one single line: 

     # aedtapp.create_lna_schematic_from_snp(input_file=tsfile_path)



But if your goal is create IPorts on each pin of the NPort component, you can use this method, which internally is doing something similar, but you will have it in one single line (create_lna_schematic_from_snp):
```
    with Desktop(specified_version="2023.1", non_graphical=False, new_desktop_session=True, close_on_exit=True):
        aedtapp = Circuit("AutomatedProject")

        work_dir = os.getcwd()
        tsfile_path = os.path.join(work_dir, tsfile)
        filename = tsfile.split(".")[0]

        **aedtapp.create_lna_schematic_from_snp(input_file=tsfile_path)**

        #----- Release AEDT instance
        aedtapp.release_desktop(close_projects=True,close_desktop=True)
```

Please let me know if this solves your questions.

**Reply by homma26153190** [reply]
Hello Samuel-san,
Thank you for your kind comment! It was very helpful. After modifying the code based on your suggestion, it now works as intended. Thank you very much.
